libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/memcpy_async.h

```diff
@@ -0,0 +1,525 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX__CUDA_MEMCPY_ASYNC_H
+#define _LIBCUDACXX__CUDA_MEMCPY_ASYNC_H
+
+#include "../cstdlib"
+
+#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+struct __single_thread_group {
+    _LIBCUDACXX_INLINE_VISIBILITY
+    void sync() const {}
+    _LIBCUDACXX_INLINE_VISIBILITY
+    constexpr _CUDA_VSTD::size_t size() const { return 1; };
+    _LIBCUDACXX_INLINE_VISIBILITY
+    constexpr _CUDA_VSTD::size_t thread_rank() const { return 0; };
+};
+
+template<_CUDA_VSTD::size_t _Alignment>
+struct aligned_size_t {
+    static constexpr _CUDA_VSTD::size_t align = _Alignment;
+    _CUDA_VSTD::size_t value;
+    _LIBCUDACXX_INLINE_VISIBILITY
+    explicit aligned_size_t(size_t __s) : value(__s) { }
+    _LIBCUDACXX_INLINE_VISIBILITY
+    operator size_t() const { return value; }
+};
+
+// Type only used for logging purpose
+enum async_contract_fulfillment {
+    none,
+    async
+};
+
+enum class __space {
+    __local,
+    __shared,
+    __cluster,
+    __global,
+    __constant,
+    __grid_constant
+};
+
+template<__space _Sp>
+using __space_constant = _CUDA_VSTD::integral_constant<__space, _Sp>;
+
+_LIBCUDACXX_DEVICE
+bool __is_cluster_shared(const void * __p) {
+    NV_IF_TARGET(NV_PROVIDES_SM_90,
+        (return __isClusterShared(__p);),
+        (return false;)
```

Comment:
We might need to add a void cast for the argument to avoid compiler warnings
```suggestion
        ((void)__p; return false;)
```